src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -110,32 +107,26 @@ class SourcesTree extends Component<Props, State> {
     if (
       projectRoot != nextProps.projectRoot ||
       debuggeeUrl != nextProps.debuggeeUrl ||
-      thread != nextProps.thread ||
       nextProps.sourceCount === 0
     ) {
       // early recreate tree because of changes
       // to project root, debugee url or lack of sources
       return this.setState(
         createTree({
-          sources: nextProps.sources[nextProps.thread],
+          sources: nextProps.sources,
```

Comment:
i think it is nice to keep the sources tree thread free as for the most part the tree should not know about other threads